sklearn/manifold/locally_linear.py

```diff
@@ -289,7 +289,14 @@ def locally_linear_embedding(
         raise ValueError("unrecognized method '%s'" % method)
 
     nbrs = NearestNeighbors(n_neighbors=n_neighbors + 1, n_jobs=n_jobs)
+
     nbrs.fit(X)
+
+    if isinstance(X, (BallTree, KDTree, NearestNeighbors)) and \
+       not (X.metric == 'euclidean'
```

Comment:
I did it because according to the specification of the Nearest Neighbors, the effective metric is only set after you call the fit method, which might not be the case since I now changed the verification to before the fit method (to save computation).